crates/ruff_graph/src/collector.rs

```diff
@@ -32,6 +32,26 @@ impl<'a> Collector<'a> {
 }
 
 impl<'ast> SourceOrderVisitor<'ast> for Collector<'_> {
+    fn enter_node(&mut self, node: AnyNodeRef<'ast>) -> TraversalSignal {
+        if self.string_imports {
+            TraversalSignal::Traverse
+        } else if matches!(
+            node,
+            AnyNodeRef::ModModule(_)
+                | AnyNodeRef::StmtFunctionDef(_)
+                | AnyNodeRef::StmtClassDef(_)
+                | AnyNodeRef::StmtWhile(_)
+                | AnyNodeRef::StmtFor(_)
+                | AnyNodeRef::StmtWith(_)
+                | AnyNodeRef::StmtIf(_)
+                | AnyNodeRef::StmtTry(_)
```

Comment:
Can we also skip over all expressions?

Reply:
Yeah this already does, right? We're skipping over everything except compound statements.